### PR TITLE
Use useAsync instead of useEffect

### PR DESCRIPTION
**Pull request by tjennison-google**

### CODE REVIEW

File: ui/src/loading.tsx [hunk 1]
```diff
@@ -1,26 +1,65 @@
...
 
-export default function Loading() {
+type State = {
+  error?: Error;
+  isPending: boolean;
+  reload: () => void;
+};
+
+type Props = {
+  state?: State;
```
Comment (calbach) [inline ui/src/loading.tsx:14]: I would avoid having a prop called `state`. I realize you probably don't mean to imply "React state" with this naming, I would guess most readers make that assumption. In general I would not recommend using the term state in the vicinity of a React component unless it's actually referring to React state.

Something like "status" maybe.

File: ui/src/loading.tsx [hunk 2]
```diff
@@ -1,26 +1,65 @@
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
 import CircularProgress from "@mui/material/CircularProgress";
-import Grid from "@mui/material/Grid";
-import React, { useEffect, useRef, useState } from "react";
+import Typography from "@mui/material/Typography";
+import React, { ReactNode, useEffect, useRef, useState } from "react";
 
-export default function Loading() {
+type State = {
```
Comment (calbach) [inline ui/src/loading.tsx:7]: opt: I don't know if it's necessary to split out this definition into its own type. I haven't done too much experimentation in this area, but pretty sure you can just inline the object type into Props below, if you preferred that.

File: ui/src/loading.tsx (hunk 1 above)
Comment (tjennison-google) [reply ui/src/loading.tsx]: Done.

File: ui/src/loading.tsx (hunk 2 above)
Comment (tjennison-google) [reply ui/src/loading.tsx]: I'm using it below for showStatus.

**Review by calbach — APPROVED** [review]

**Pull request merged**